packages/nimble/R/BUGS_getDependencies.R

```diff
@@ -62,3 +62,310 @@ gd_getDependencies_IDs <- function(graph, maps, nodes, omit, downstream) {
+        ## Then update the explainDerivContent function.  May also make more sense to make this function a ref class w/ fields for different return vals and such.
+        lineWRTArgsAsCharacters <<- list()
+        lineWRTArgSizeInfo <<- list()
+        calcWithArgsCalls  <<- list()
+        for(i in seq_along(depIndex_2_parentDepIndices)){
+          if(calcNodeIndicators[i] == 1){
+            lineWRTArgSizeInfo[[i]]      <<- numeric(length(depIndex_2_parentDepIndices[[i]]))
+            sizeAndDimInfo <- environment(model$nodeFunctions[[declIDs[i]]]$.generatorFunction)[['parentsSizeAndDims']]
+            formalArgNames <- formals(model$nodeFunctions[[declIDs[i]]]$calculateWithArgs)
+            unrolledIndicesMatrixRow <- model$modelDef$declInfo[[declIDs[i]]]$unrolledIndicesMatrix[ unrolledIndicesMatrixRows[i], ]
+            modelArgNames <- lapply(names(formalArgNames)[-1],
+                                    function(x){parse(text = convertCalcArgNameToModelNodeName(x, sizeAndDimInfo, unrolledIndicesMatrixRow))[[1]]})
+            calcWithArgsCalls[[i]] <<- as.call(c(list(as.name('calcWithArgs'), unrolledIndicesMatrixRow), modelArgNames))
+            for(j in seq_along(depIndex_2_parentDepIndices[[i]])){
+              if(j == 1 && WRTNodeIndicators[[i]] == 1){
+                thisWrtLine <- which(WRTLineNums == i)
+                lineWRTArgsAsCharacters[[i]]  <<- names(formalArgNames[2])
+                lineWRTArgSizeInfo[[i]][1]  <<- WRTLineSize[[thisWrtLine]]
+              }
+              else if(depIndex_2_parentDepIndices[[i]][[j]][1] > 0){
+                for(k in 1:length(depIndex_2_parentDepIndices[[i]][[j]])){
+                  wrtInfoList <-  convertToWrtArg(allWRTAndCalcNodeNames[depIndex_2_parentDepIndices[[i]][[j]][k]],
+                                                  modelArgNames[[j]],
+                                                  names(formalArgNames)[j+1],
+                                                  thisModel)
+                  if(length(lineWRTArgsAsCharacters) < i){
+                    lineWRTArgsAsCharacters[[i]] <<- wrtInfoList$wrtArg
+                  }
+                  else{
+                    lineWRTArgsAsCharacters[[i]] <<- c(lineWRTArgsAsCharacters[[i]], wrtInfoList$wrtArg)
+                  }
+                  lineWRTArgSizeInfo[[i]][j] <<- lineWRTArgSizeInfo[[i]][j] + wrtInfoList$argSize
+                }
+              }
+            }
+          }
+          else{
+            lineWRTArgsAsCharacters[[i]] <<- NA
+            lineWRTArgSizeInfo[[i]]      <<- NA
+            calcWithArgsCalls[[i]] <<-   NA
+          }
+          if(length(lineWRTArgsAsCharacters) < i){
+            lineWRTArgsAsCharacters[[i]] <<- NA
+          }
+        }
+      },
+
+      # ### A function that substitutes correct values of unrolledIndicesMatrix
+      # ### into symbolicParentNodesReplaced.
+      recurseReplaceIndices = function(code, unrolledIndicesRow){
```

Comment:
2 questions:
- could this function be a non-method?
- would `substitute` work?  Is the goal to replace in `code` the values by name in `unrolledIndicesRow`?
```{r}
vals <- data.frame(a = 1, b = 2)
code <- quote(x[a, k[b]])
substitute(code, as.list(vals))
eval(substitute(substitute(code, as.list(vals)), list(code = code)))
```

I am not sure if I have captured the idea correctly.

Reply:
Yes, `substitute` works perfectly here, so the method can be removed.  Thanks for the suggestion!